### PR TITLE
OAuth2 and security related improvements

### LINKED ISSUE #86: OAuth token refresher should not require access token URL environment variable
See [here](https://github.com/zalando/friboo/blob/master/src/org/zalando/stups/friboo/system/oauth2.clj#L23). This makes local testing hard. The underlying `tokens` library allows to set the tokens to use [with an environment variable](https://github.com/zalando/tokens#local-testing). Friboo should respect this 😄

### LINKED ISSUE #89: Environment is read incorrectly when loading configuration
When running on Linux, there is an environment variable `DBUS_SESSION_BUS_ADDRESS` that gets picked up as part of `:db` namespace. We need to only pick up variables starting with `DB_`, not just `DB` in this case.

### PR DESCRIPTION
**Pull request by dryewo**
- Fix #86 (allow static tokens for testing via `OAUTH2_ACCESS_TOKENS`, don't require `OAUTH2_ACCESS_TOKEN_URL`).
- Fix #89 (To qualify as namespace, require env var prefix to end with `_` when loading configuration) .
- Don't let Tokens library use `CREDENTIALS_DIR` env var directly, to support reloaded workflow.
- Separate incoming token checking functions and service token refreshing component into different namespaces.
- Improve configuration loading to support arbitrary mappings, not only `TOKENINFO_URL -> HTTP_TOKENINFO_URL` and `CREDENTIALS_DIR -> OAUTH2_CREDENTIALS_DIR`

### CODE REVIEW

File: src/org/zalando/stups/friboo/security.clj [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+  (api/error 401 "Unauthorized"))
+
+(defn- forbidden
+  "Send forbidden response."
+  [reason]
+  (log/warn "ACCESS DENIED (forbidden) because %s." reason)
+  (api/error 403 "Forbidden"))
+
+(defn allow-all
+  "Returns a swagger1st security hander that allows everything."
```
Comment (harti2006) [inline src/org/zalando/stups/friboo/security.clj]: handler

File: src/org/zalando/stups/friboo/system/oauth2.clj [hunk 2]
```diff
@@ -1,131 +1,107 @@
...
-              (assoc request :tokeninfo tokeninfo)
-              (forbidden "scopes not granted"))
-            (unauthorized "invalid access token"))
-          (api/error 503 "token info misconfigured")))
-      (unauthorized-without-logging "no access token given"))))
\ No newline at end of file
+  [token-id {:keys [token-storage static-tokens]}]
+  (if-let [static-token (get static-tokens token-id)]
+    static-token
+    (when token-storage
```
Comment (harti2006) [inline src/org/zalando/stups/friboo/system/oauth2.clj]: throw AccessTokenUnavailable exception if key does not exist

**Comment by harti2006** [comment]
:+1:

**Comment by dryewo** [comment]
👍

**Pull request merged**